packages/components/src/viewport/aware/index.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+import React from 'react';
+import { View } from 'react-native';
+import * as Utils from '../utils'
+
+export default WrappedComponent => {
+  return class extends React.Component {
+    constructor(props, context) {
+      super(props, context);
+      this.state = {
+        mounted: false,
+        inViewport: false,
+      };
+    }
+
+    componentDidMount() {
+      this.context.addViewportListener
+        && this.context.addViewportListener(this._onViewportChange);
+      this.setState({
```

Comment:
Two issues:
1. wrapperRef != null <=> this.state.ismounted, you can remove the flag
2. using the "official" state to like this causes unnecessary re-renders; it would have been better to use a direct attribute on `this`.... but you don't need it

Reply:
Yes, you are right that `state.isMounted` is not needed. Thanks for the tip regarding the "official" state!